docs/DeveloperGuide.md

```diff
@@ -2,37 +2,186 @@
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+Referred to https://se-education.org/addressbook-level3/DeveloperGuide.html when drafting the DG
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Architecture
 
+![](./images/architeture.png)
+
+The Architecture Diagram given above explains the high-level design of the App.
+
+The **User** sends command which is processed by Input
+
+**Input**: Receives user input string, processes it and sends to CashLeh main app
+
+**CashLeh**: main app
+
+**UI**: Responsible for the printing onto CLI
+
+**Parser**: Process the input string and calls respective methods
+
+**FileStorage**: Reads data from, and writes data to file
+
+### How the architecture components interact with each other
+
+![](./images/main_example.png)
+
+The Sequence Diagram above shows how the components interact with each other when the user enters a command
+
+### Budget Handler
+
+![](./images/budget.png)
+
+### String Tokenizer
+
+The String Tokenizer assists in the parsing of user input. It is used to split the user input into a hashmap according to the delimiter/prefix of each command. This allows the developer to more easily customize the commands and add new commands without having to worry about the parsing of the user input.
+
+> **Example**:\
+> `addIncome money /amt 1000 /date 2021-10-10` with prefixes array `{"addIncome", "/amt", "/date"}` will be tokenized into a hash map with the following key-value pairs:\
+> `addIncome: money`\
+> `amt: 1000`\
+> `date: 2021-10-10`
+
+Delimiters can also be specified as optional by adding ":optional" to the end. This allows the user to omit the prefix when entering the command.
+
+#### Implementation
+
+The tokenize function is implemented by the StringTokenizer class. It takes in the user input and the prefixes array and returns a hashmap of the user input.
+
+![](./images/stringTokenizer.png)
```

Comment:
The return arrow should be dotted.
<img width="631" alt="image" src="https://github.com/nus-cs2113-AY2324S1/tp/assets/142559789/f3d51fd4-018e-49b3-9c98-591d159c0f02">


Lifeline should not go past "X"
<img width="341" alt="Screenshot 2023-11-01 at 11 23 11 AM" src="https://github.com/nus-cs2113-AY2324S1/tp/assets/142559789/61f220cd-5e66-448d-bc08-107077224074">